scripts/upgradeable/transpile-onto.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if [ $# -lt 1 ]; then
+  echo "usage: bash $0 <target> [base]" >&2
+  exit 1
+fi
+
+set -x
+
+target="$1"
+base="${2-}"
+
+bash scripts/upgradeable/transpile.sh
+
+commit="$(git rev-parse --short HEAD)"
+branch="$(git rev-parse --abbrev-ref HEAD)"
+
+git add contracts
+
+git checkout --quiet --detach
+
+if git rev-parse -q --verify "$target"; then
+  git reset --soft "$target"
+  git checkout "$target"
+else
+  git checkout --orphan "$target"
+  if [ -n "$base" ] && git rev-parse -q --verify "$base"; then
+    git reset --soft "$base"
+  fi
```

Comment:
I have the feeling that we should add an `else` here if there's no `base` and `target` doesn't exist. My concern is that the next check will fail because the transpiler runs `patch-apply.sh`, adding changes:

```bash
if ! git diff --quiet --cached; then
  git commit -m "Transpile $commit"
fi
```

Reply:
I'm not sure I understand the comment/concern.

I've pushed a commit adding some comments in this file. Let me know if those clear things up.

Reply:
My concern is that the transpile will add changes even if there's no `target` or `base`, in other words, there's no `else` case in which we fail (`exit 1`) from the script. In such case, we'll commit without the actual transpilation and then push to the `contracts-upgradeable` with a commit not including the upgradeable contracts.

Note that it'll only happen if we don't provide `target` or `base`, which can happen due to a bug or human error changing the scripts. Right now it works so I'll suggest to just add a comment, but I wouldn't block the merge since we know it works as of now.